### PR TITLE
OKTA-596621 Add SSPR primary factor and authenticator key constraints to Password Policy

### PR DESCRIPTION
**Pull request by vanngo-okta**
## Description:
- **What's changed?** Add Password Policy changes: SSPR authenticators and constraints. This PR needs to be aligned with PR 404 in okta-oas3 repo.
- **Is this PR related to a Monolith release?** TBD (2023.06.0?)

### Preview:
https://643474c17272e412a24cf56d--reverent-murdock-829d24.netlify.app/docs/reference/api/policy/#password-rules-action-data
image

image

image

image

image

### Resolves:

* [OKTA-596621](https://oktainc.atlassian.net/browse/OKTA-596621)

### CODE REVIEW

**Review by antonioruberto-okta — APPROVED** [review]
lg2m

**Review by markfreed-okta — APPROVED** [review]
🚢

File: packages/@okta/vuepress-site/docs/reference/api/policy/index.md [hunk 1]
```diff
@@ -1790,16 +1790,63 @@ In the final example, end users are required to verify two Authenticators before
...
+###### Self Service Password Reset Action Primary Requirement object
+
+<ApiLifecycle access="ie" />
+
+Defines the authenticators permitted for the initial authentication step of password recovery
+
+| Property | Description | Data Type | Supported Values | Required | Default |
+| -------- | ----------- | --------- | ---------------- | -------- | ------- |
+| methodConstraints | Specifies an authenticator-specific constraint on the values in the `methods` array. Currently, Google OTP is the only accepted constraint for the `OTP` method. | Array of [Self Service Password Reset Action Primary Requirement Constraint object](#self-service-password-reset-action-primary-requirement-constraint-object) | `[ {"method": "otp", "allowedAuthenticators": [ { "key": "google_otp" } ] } ]` | No | |
+| methods | Authenticator methods allowed for the initial authentication step of password recovery. The `OTP` method requires a constraint limiting it to the Google authenticator. | Array | `PUSH`, `SMS`, `VOICE`, `EMAIL`, `OTP` | No | |
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/api/policy/index.md:1823]:
```suggestion
| methods | Authenticator methods allowed for the initial authentication step of password recovery. The `OTP` method requires a constraint limiting it to the Google authenticator. | Array | `PUSH`, `SMS`, `VOICE`, `EMAIL`, and `OTP` | No | |
```

File: packages/@okta/vuepress-site/docs/reference/api/policy/index.md [hunk 2]
```diff
@@ -1790,16 +1790,63 @@ In the final example, end users are required to verify two Authenticators before
...
+
+###### Self Service Password Reset Action Primary Requirement Constraint object
+
+<ApiLifecycle access="ie" />
+
+Constraints on the values specified in the `selfServicePasswordReset.requirement.primary.methods` array. Specifying a constraint limits methods to specific authenticators. Currently, Google OTP is the only accepted constraint.
+
+| Property | Description | Data Type | Supported Values | Required | Default |
+| -------- | ----------- | --------- | ---------------- | -------- | ------- |
+| allowedAuthenticators | Limits the authenticators that can be used for a given method. Currently, only the `OTP` method supports constraints and Google authenticator is the only allowed authenticator. | Array of [authenticator keys](/docs/reference/api/authenticators-admin/#authenticator-properties). | `[ { "key": "google_otp" } ]` | No | |
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/api/policy/index.md:1833]:
```suggestion
| allowedAuthenticators | Limits the authenticators that can be used for a given method. Currently, only the `OTP` method supports constraints and Google authenticator is the only allowed authenticator. | Array of [authenticator keys](/docs/reference/api/authenticators-admin/#authenticator-properties) | `[ { "key": "google_otp" } ]` | No | |
```

File: packages/@okta/vuepress-site/docs/reference/api/policy/index.md [hunk 3]
```diff
@@ -1790,16 +1790,63 @@ In the final example, end users are required to verify two Authenticators before
...
+
+<ApiLifecycle access="ie" />
+
+Defines the secondary authenticators needed for password reset if `selfServicePasswordReset.requirement.stepUp.required` is true. The following are three valid configurations:
+
+* `required` = false
+* `required` = true with no methods to use any SSO authenticator
+* `required` = true with `security_question` as the method
+
+| Property | Description | Data Type | Supported Values | Required | Default |
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/api/policy/index.md:1846]: Is there a default value for these rows? Maybe we don't need this column?

**Review by brianduffield-okta — APPROVED** [review]
LGTM ... couple nits and one question

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/policy/index.md](https://github.com/okta/okta-developer-docs/blob/1b5fb86174c51ba96435ef75c552912f0f691376/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fpolicy%2Findex.md) | [69](https://okta.acrolinx.cloud/api/v1/checking/scorecards/07f8a4ce-115f-4568-8304-b682e4767275) | 62  | 71  | 60  | 85  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.50a0b220-3602-4f8d-af42-569ca7b53364?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/reference/api/policy/index.md (hunk 3 above)
Comment (vanngo-okta) [reply packages/@okta/vuepress-site/docs/reference/api/policy/index.md:1846]: Nope, no default values. I'll remove this column in another commit. Thanks for the suggestion!

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/reference/api/policy/index.md](https://github.com/okta/okta-developer-docs/blob/25b23acf39a9217e1bafad48c10595e754969a84/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Freference%2Fapi%2Fpolicy%2Findex.md) | [69](https://okta.acrolinx.cloud/api/v1/checking/scorecards/df7029d5-6c15-47c9-a61e-93beac4e46c4) | 62  | 71  | 60  | 85  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.04e842ce-8bbb-48fd-8fdc-d7873746fac8?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Pull request merged**